pkg/volume/emptydir/empty_dir.go

```diff
@@ -229,13 +231,25 @@ func (ed *emptyDir) SetUpAt(dir string, fsGroup *int64) error {
 		err = fmt.Errorf("unknown storage medium %q", ed.medium)
 	}
 
-	volume.SetVolumeOwnership(ed, fsGroup)
+	volume.SetVolumeOwnership(ed, mounterArgs.FsGroup)
 
 	if err == nil {
 		volumeutil.SetReady(ed.getMetaDir())
 	}
-
-	return err
+	if mounterArgs.DesiredSize != nil {
+		hasQuotas, err := quota.SupportsQuotas(ed.mounter, dir)
+		if err != nil {
+			return fmt.Errorf("Unable to check for quota support on %s: %s", dir, err.Error())
+		}
+		if hasQuotas {
+			klog.V(4).Infof("emptydir trying to assign quota %v on %s", mounterArgs.DesiredSize, dir)
+			err := quota.AssignQuota(ed.mounter, dir, mounterArgs.PodUID, mounterArgs.DesiredSize)
+			if err != nil {
+				return fmt.Errorf("Set quota on %s failed %s", dir, err.Error())
```

Comment:
if we return error here, we might have the following problem.
1. mount succeeded
2. assignQuota failed
3. pod is deleted
4. mount would be remained because reconciler has no record of this mount.

I am thinking two options here
1. Ignore the error, log it or submit event for user
2. before returning error, try to unmount it. But unmount might fail even after a few tries. We might still has some mounts left, although it should corner cases.